### PR TITLE
Refactor to use only kernel-modules-headers

### PR DESCRIPTION
**Pull request by alexgg**
* Add LICENSE
* Use multi-container docker compose
* Simplify build script:
  * Pull headers directly from path instead of using S3 bucket parsing
  * Use only `kernel-modules-headers` instead of multiple redundat header files.
  * Always run `make modules_prepare`
  * Remove list and build functionality - just build the modules
  * Auto-detect the correct path stripping
  * Separate logging functions to a different scritp
* Provide a defaulr entry.sh script that checks for the example kernel module output

TODO: convert into a block
Change-type: major

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        # Modify to the desired balenaOS version
+        OS_VERSION: 2.108.27
+    privileged: true
+    restart: on-failure
+  check:
+    build:
+      context: ./check
+      dockerfile: Dockerfile.template
+    depends_on:
+      - load
```
Comment (klutchell) [inline docker-compose.yml:18]: If we renamed this file to `docker-compose.test.yml` and changed the check service to `sut` would Flowzone be able to test this?

File: module/build.sh [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+		-h Display usage
+EOF
+}
+
+fetch_headers()
+{
+	local slug="${1}"
+	local version="${2}"
+	local files_url="https://files.balena-cloud.com"
+	local esr_pattern="^[1-3][0-9]{3}\.(1|01|4|04|7|07||10)\.[0-9]*\.?(dev|prod)?$"
```
Comment (klutchell) [inline module/build.sh]:
```suggestion
	local esr_pattern="^[1-3][0-9]{3}\.(1|01|4|04|7|07||10)\.[0-9]*(\.dev|\.prod)?$"
```

File: docker-compose.yml (hunk 1 above)
Comment (alexgg) [reply docker-compose.yml:18]: I'll try..

Comment (alexgg) [reply docker-compose.yml:18]: I tried to make this work but there is no way to have the test work as our `generic-amd64` builds will never match  the Ubuntu version used in the docker test.

**Comment by alexgg** [comment]
lgtm

**Pull request merged**